### PR TITLE
envoy: 3876d7c0f

### PR DESCRIPTION
**Pull request by junr03**
Includes three relevant commits:
https://github.com/envoyproxy/envoy/pull/17226
https://github.com/envoyproxy/envoy/pull/17274
https://github.com/envoyproxy/envoy/pull/17207

### CODE REVIEW

File: library/common/config/config.cc [hunk 1]
```diff
@@ -183,6 +183,7 @@ const char* config_template = R"(
               dns_failure_refresh_rate:
                 base_interval: *dns_fail_base_interval
                 max_interval: *dns_fail_max_interval
+              dns_query_timeout: 25s
```
Comment (junr03) [inline library/common/config/config.cc:186]: hard coding this in this current PR so no one is burnt by the default. I am doing a PR to make it configurable.

File: library/common/config/config.cc [hunk 2]
```diff
@@ -36,6 +36,7 @@ const std::string config_header = R"(
 - &dns_refresh_rate 60s
 - &dns_fail_base_interval 2s
 - &dns_fail_max_interval 10s
+- &dns_query_timeout 25s
```
Comment (junr03) [inline library/common/config/config.cc:39]: Putting a default so no one is burnt by the upstream 5s default. I am working on a parallel PR to make it configurable.

File: library/common/config/config.cc [hunk 3]
```diff
@@ -36,6 +36,7 @@ const std::string config_header = R"(
 - &dns_refresh_rate 60s
 - &dns_fail_base_interval 2s
```
Comment (junr03) [inline library/common/config/config.cc:37]: @goaway in slack we discussed 25s timeout + 5s failure refresh rate. The rate as is here starts a 2s and backoffs to 10s. Are we ok with that policy or do you want to reduce max to 5s?

File: library/common/config/config.cc (hunk 2 above)
Comment (junr03) [reply library/common/config/config.cc:39]: https://github.com/envoyproxy/envoy-mobile/pull/1583

**Review by rebello95 — APPROVED** [review]

File: library/common/config/config.cc (hunk 3 above)
Comment (goaway) [reply library/common/config/config.cc:37]: I think that policy is fine for now.

**Pull request merged**